internal/monitor/controller.go

```diff
@@ -249,8 +276,15 @@ func (cm *PodMonitorType) controllerCleanupPod(pod *v1.Pod, node *v1.Node, reaso
 
 	// Call the driver to validate the volumes are not in use
 	if cm.CSIExtensionsPresent && CSIApi.Connected() {
-		log.WithFields(fields).Infof("Validating host connectivity for node %s volumes %v", node.ObjectMeta.Name, volIDs)
+		log.WithFields(fields).Infof("Checking host connectivity for node %s and iosInprogress for volumes %v", node.ObjectMeta.Name, volIDs)
 		connected, iosInProgress, err := cm.callValidateVolumeHostConnectivity(node, volIDs, true)
+		log.WithFields(fields).Infof("Validating host connectivity for node %s volumes %v connected %t iosInProgress %t", node.ObjectMeta.Name, volIDs, connected, iosInProgress)
+		// If the volume's access mode is RWX, ignore iosInProgress, as other applications may perform I/O operations on the volume.
+		status := isRWXVolume(pvlist)
+		if status {
```

Comment:
suggestion: no need to assign the value of the check to `status` unless `status` is used elsewhere? This also has the added benefit of making it more readable.
```suggestion
		if isRWXVolume(pvlist) {
```